### ISSUE TITLE
False-positive while_immutable_condition lint when variable is struct field

### ISSUE THREAD

**Issue by bvinc** [rust-lang-nursery/rust-clippy#2631]
```rust

pub struct Foo {
    foo: u64,
}

impl Foo {
    fn bar(&mut self) {
        while self.foo < 100 {
            self.foo += 1;
        }
    }
}

fn main() {
}
```

Warning:
```
error: Variable in the condition are not mutated in the loop body. This either leads to an infinite or to a never running loop.
 --> src/main.rs:8:15
  |
8 |         while self.foo < 100 {
  |               ^^^^^^^^^^^^^^
  |
  = note: #[deny(while_immutable_condition)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.191/index.html#while_immutable_condition

error: aborting due to previous error
```
